src/atomate2/vasp/jobs/defect.py

```diff
@@ -0,0 +1,182 @@
+"""Definition of defect job maker."""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import Iterable, List
+
+from jobflow import Flow, Maker, Response, job
+from pydantic import BaseModel
+from pymatgen.core import Structure
+from pymatgen.io.vasp import Incar
+from pymatgen.io.vasp.outputs import WSWQ
+
+from atomate2.common.files import copy_files, gunzip_files, gzip_files, rename_files
+from atomate2.utils.file_client import FileClient
+from atomate2.utils.path import strip_hostname
+from atomate2.vasp.files import copy_vasp_outputs
+from atomate2.vasp.jobs.core import StaticMaker
+from atomate2.vasp.run import run_vasp
+from atomate2.vasp.schemas.defect import CCDDocument, FiniteDifferenceDocument
+from atomate2.vasp.schemas.task import TaskDocument
+
+logger = logging.getLogger(__name__)
+
+
+class CCDInput(BaseModel):
+    """Document model to help construct CCDDocument."""
+
+    structure: Structure
+    energy: float
+    dir_name: str
+    uuid: str
+
+
+@job
+def spawn_energy_curve_calcs(
+    ref: Structure,
+    distorted: Structure,
+    distortions: Iterable[float],
+    static_maker: StaticMaker,
+    prev_vasp_dir: str | Path | None = None,
+    add_name: str = "",
+):
+    """
+    Compute the total energy curve as you distort a reference structure to a distorted structure.
+
+    Parameters
+    ----------
+    ref : pymatgen.core.structure.Structure
+        pymatgen structure corresponding to the ground (final) state
+    distorted : pymatgen.core.structure.Structure
+        pymatgen structure corresponding to the excited (initial) state
+    static_maker : atomate2.vasp.jobs.core.StaticMaker
+        StaticMaker object
+    distortions : tuple
+        list of distortions to apply
+
+    Returns
+    -------
+    Response
+        Response object
+    """
+    jobs = []
+    outputs = []
+
+    # add the static job for the reference structure
+    static_maker.make(ref)
+
+    distorted_structures = ref.interpolate(distorted, nimages=sorted(distortions))
+    # add all the distorted structures
+    for i, d_struct in enumerate(distorted_structures):
+        static_job = static_maker.make(d_struct, prev_vasp_dir=prev_vasp_dir)
+        suffix = f" {i+1}" if add_name == "" else f" {add_name} {i}"
+        static_job.append_name(f"{suffix}")
```

Comment:
Similar to what I did in the electron phonon workflow, can you also add some metadata that will get written to a json file that includes the distortion value, charge state, and reference structure? Any json files will automatically get parsed by the TaskDocument and they're always useful to have in the database later on.

E.g., see what I did here: https://github.com/materialsproject/atomate2/blob/525d0ff78e880fc9c17aa8a9597ba810444681e2/src/atomate2/vasp/jobs/elph.py#L167-L178

Using the `update_maker_kwargs` is necessary in case someone uses a flow maker instead of a job maker for the static calculations, even if the static calculation is a job in this version of the workflow. One reason for this could be you want to do a gamma point only calculation first and then 2x2x2 afterwards. All you'd need to do is replace the `static_maker` with a flow maker and nothing else in the workflow has to change.